### PR TITLE
Improve clear_not_launched_queued_tasks call duration

### LINKED ISSUE #34877: Scheduler is spending most of its time in clear_not_launched_queued_tasks function
### Apache Airflow version

2.7.1

### What happened

Airflow running the clear_not_launched_queued_tasks function on a certain frequency (default 30 seconds). When we run the airflow on a large Kube cluster (pods more than > 5K).  Internally the clear_not_launched_queued_tasks function loops through each queued task and checks the corresponding worker pod existence in the Kube cluster. Right this existence check using list pods Kube API. The API is taking more than 1s. if there are 120 queued tasks, then it will take ~ 120 seconds (1s * 120). So, this leads the scheduler to spend most of its time in this function rather than scheduling the tasks. It leads to none of the jobs being scheduled or degraded scheduler performance.

### What you think should happen instead

It would be nice to get all the airflow worker pods in a one/few batch calls rather than for each task. These batch calls helps to speed the processing of clear_not_launched_queued_tasks function call. 

### How to reproduce

Run the airflow on large Kube clusters (> 5K pods). Simulate the airflow to run the 100 parallel DAG runs for every minute. 

### Operating System

Cent OS 7

### Versions of Apache Airflow Providers

2.3.3, 2.7.1

### Deployment

Other Docker-based deployment

### Deployment details

Terraform based airflow deployment

### Anything else

_No response_

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by dirrao**
Problem: Airflow running the clear_not_launched_queued_tasks function on a certain frequency (default 30 seconds). When we run the airflow on a large Kube cluster (pods more than > 5K). Internally the clear_not_launched_queued_tasks function loops through each queued task and checks the corresponding worker pod existence in the Kube cluster. Right this existence check using list pods Kube API. The API is taking more than 1s. if there are 120 queued tasks, then it will take ~ 120 seconds (1s * 120). So, this leads the scheduler to spend most of its time in this function rather than scheduling the tasks. It leads to none of the jobs being scheduled or degraded scheduler performance.

Solution: Use k8 list pods batch api call to get all the worker pod owned by scheduler. Prepare the set of searchable strings using pod labels. Use this set data structure and identify whether the task associated pod exists or not.

set elements string format:
(dag_id=<dag_id>,task_id=<task_id>,airflow-worker=[,map_index=<map_index>],[run_id=<run_id>]|[execution_date=<execution_date>])

The details for the issue is mentioned in the below ticket https://github.com/apache/airflow/issues/34877

Closes: #34877

### CODE REVIEW

**Comment by uranusjr** [comment]
Please consider reformatting the commit message so it fits the standard style.

https://cbea.ms/git-commit/#seven-rules

File: airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py [hunk 1]
```diff
@@ -224,6 +224,36 @@ def clear_not_launched_queued_tasks(self, session: Session = NEW_SESSION) -> Non
             if time.time() - timestamp > allowed_age:
                 del self.last_handled[key]
 
+        if len(queued_tis) == 0:
```
Comment (uranusjr) [inline airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py]:
```suggestion
        if not queued_tis:
```

File: airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py [hunk 2]
```diff
@@ -224,6 +224,36 @@ def clear_not_launched_queued_tasks(self, session: Session = NEW_SESSION) -> Non
...
+            return
+
+        # airflow worker label selector batch call
+        kwargs = dict(label_selector=f"airflow-worker={self._make_safe_label_value(str(self.job_id))}")
```
Comment (uranusjr) [inline airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py]:
```suggestion
        kwargs = {"label_selector": f"airflow-worker={self._make_safe_label_value(str(self.job_id))}"}
```

File: airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py [hunk 3]
```diff
@@ -224,6 +224,36 @@ def clear_not_launched_queued_tasks(self, session: Session = NEW_SESSION) -> Non
...
+        if self.kube_config.kube_client_request_args:
+            kwargs.update(**self.kube_config.kube_client_request_args)
```
Comment (uranusjr) [inline airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py]:
```suggestion
            kwargs.update(self.kube_config.kube_client_request_args)
```

`dict.update()` can take a dict directly without unpacking.

**Review by hussein-awala — COMMENTED** [review]
This PR will slightly increase memory usage, especially if parallelism is set to a high value, but it will have a significant positive impact on scheduling time and scheduler performance.

Overall it looks good, but I prefer to test it first, I will try to do that ASAP.

**Comment by dirrao** [comment]
@hussein-awala Did you get a chance to test it? Any findings?

**Review by hussein-awala — APPROVED** [review]
I re-checked the code, and I tested the change with Breeze and with a production deployment with 32 pods per scheduler; all looks good.

**Comment by dirrao** [comment]
@uranusjr / @jedcunningham,
can you review it?

**Review by eladkal — APPROVED** [review]

**Pull request merged**